### ISSUE TITLE
False positive no-name-in-module for ProcessPoolExecutor in python 3.14

### ISSUE THREAD

**Issue by kasium** [PyCQA/pylint#10632]
### Bug description
The following code leads to issues only in python 3.14

```python
from concurrent.futures import ProcessPoolExecutor
```

### Command used

```shell
pylint x.py
```

### Pylint output

```python
************* Module x
x.py:1:0: C0114: Missing module docstring (missing-module-docstring)
x.py:3:0: E0611: No name 'ProcessPoolExecutor' in module 'concurrent.futures' (no-name-in-module)
x.py:3:0: W0611: Unused ProcessPoolExecutor imported from concurrent.futures (unused-import)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 3.3.9
astroid 3.3.11
Python 3.14.0 (main, Oct  9 2025, 11:33:11) [GCC 7.5.0]
```

### OS / Environment

Linux

**Comment by kasium** [comment]
I checked the relevant code in cpython and there is a difference between 3.13 and 3.14

*3.13*
```py
def __getattr__(name):
    global ProcessPoolExecutor, ThreadPoolExecutor

    if name == 'ProcessPoolExecutor':
        from .process import ProcessPoolExecutor as pe
        ProcessPoolExecutor = pe
        return pe

    if name == 'ThreadPoolExecutor':
        from .thread import ThreadPoolExecutor as te
        ThreadPoolExecutor = te
        return te

    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
```

*3.14*
```py
def __getattr__(name):
    global ProcessPoolExecutor, ThreadPoolExecutor, InterpreterPoolExecutor

    if name == 'ProcessPoolExecutor':
        from .process import ProcessPoolExecutor
        return ProcessPoolExecutor

    if name == 'ThreadPoolExecutor':
        from .thread import ThreadPoolExecutor
        return ThreadPoolExecutor

    if _interpreters and name == 'InterpreterPoolExecutor':
        from .interpreter import InterpreterPoolExecutor
        return InterpreterPoolExecutor

    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
```

I guess that the local assignment lets pylint understand whats going on
